zyte_spider_templates/_incremental/manager.py

```diff
@@ -22,11 +23,19 @@
 logger = logging.getLogger(__name__)
 
 INCREMENTAL_SUFFIX = "_incremental"
+_MAX_LENGTH = 2048 - len(INCREMENTAL_SUFFIX)
 COLLECTION_API_URL = "https://storage.scrapinghub.com/collections"
 
 THREAD_POOL_EXECUTOR = ThreadPoolExecutor(max_workers=10)
 
 
+def _get_collection_name(crawler: Crawler) -> str:
+    if name := crawler.settings.get("INCREMENTAL_CRAWL_COLLECTION_NAME"):
+        return name
+    name = get_spider_name(crawler).rstrip("_")[:_MAX_LENGTH] + INCREMENTAL_SUFFIX
+    return re.sub(r"[^a-zA-Z0-9_]", "_", name)
```

Comment:
This one may be problematic for another reason - all spiders with unicode names of the same length would silently get the same collection name, and reuse the fingerprint DB unintentionally.

Reply:
If we do not care about human readability of the collection names, we could replace characters with their UTF-8 hexadecimal (e.g. å → C3A5) or some other Unicode character ID.

Otherwise, I think we are back to slugify or at least Unidecode, with the caveat that the results may change over time since we have no control over those libraries. We could pin them in `zyte-spider-templates-project`, but eventually some user may find this problematic.

We could also use the spider numeric ID, since collections are project-specific. We can extract it easily from the middle of the job ID.

Reply:
I also wonder what are the restrictions on the spider names. The solution could also be to bring the formats closer together (e.g. add more validation for virtual spider names). 

> Otherwise, I think we are back to slugify or at least Unidecode

Hm, using text-unidecode could be pretty stable. Unlike unidecode or python-slugify, it didn't have a release in many-many years :)

Reply:
Oh, nice! And its maintainer seems familiar :)

Reply:
What do you think about just disabling the default name creation and require an explicit collection name?

Reply:
Good point. I think it is worth considering. While it is a slightly worse initial user experience, it is the most reliable approach, no need for us to worry about any long-term issues with the default name generation.

We could make it a backward-compatible change by making it required only in the UI with some JSON schema customization, and logging a deprecation error (because warnings are not visible in the UI easily) to encourage users to set it.